posts/2023-08-24-Rust-1.72.0.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Announcing Rust 1.72.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.72.0. Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, you can get 1.72.0 with:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`](https://www.rust-lang.org/install.html) from the appropriate page on our website, and check out the [detailed release notes for 1.72.0](https://github.com/rust-lang/rust/releases/tag/1.72.0) on GitHub.
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`). Please [report](https://github.com/rust-lang/rust/issues/new/choose) any bugs you might come across!
+
+## What's in 1.72.0 stable
+
```

Comment:
``````suggestion

### Rust reports potentially useful `cfg`-disabled items in errors

You can conditionally enable Rust code using `cfg`, such as to provide certain
functions only with certain crate features, or only on particular platforms.
Previously, items disabled in this way would be effectively invisible to the
compiler. Now, though, the compiler will remember the name and `cfg` conditions
of those items, so it can report (for example) if a function you tried to call
is unavailable because you need to enable a crate feature.

```
   Compiling my-project v0.1.0 (/tmp/my-project)
error[E0432]: unresolved import `rustix::io_uring`
   --> src/main.rs:1:5
    |
1   | use rustix::io_uring;
    |     ^^^^^^^^^^^^^^^^ no `io_uring` in the root
    |
note: found an item that was configured out
   --> /home/username/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.8/src/lib.rs:213:9
    |
213 | pub mod io_uring;
    |         ^^^^^^^^
    = note: the item is gated behind the `io_uring` feature

For more information about this error, try `rustc --explain E0432`.
error: could not compile `my-project` (bin "my-project") due to previous error
```

``````

Reply:
This seems like another headline user-visible feature of this release, making `cfg` easier to work with.